rtrlib/rtr/packets.c

```diff
@@ -1049,6 +1094,55 @@ static int rtr_update_spki_table(struct rtr_socket *rtr_socket, struct spki_tabl
 	return RTR_SUCCESS;
 }
 
+static int rtr_update_aspa_table(struct rtr_socket *rtr_socket, struct aspa_table *aspa_table, const void *pdu)
+{
+	const enum pdu_type type = rtr_get_pdu_type(pdu);
+
+	assert(type == ASPA);
+
+	struct aspas_record entry;
+
+	rtr_aspa_pdu_2_aspa_record(rtr_socket, pdu, &entry, type)
+
+	int rtval;
+	// TODO: are there any other flags different from 0x01?
+	if (((struct pdu_aspa *)pdu)->flags & 1 == 1) {
+		rtval = aspa_table_add_entry(aspa_table, &entry);
+
+	} else if (((struct pdu_aspa *)pdu)->flags & 1 == 0) {
+		rtval = aspa_table_remove_entry(aspa_table, &entry);
+
+	} else {
+		const char txt[] = "ASPA PDU with invalid flags value received";
+
+		RTR_DBG("%s", txt);
+		rtr_send_error_pdu_from_host(rtr_socket, pdu, pdu_size, CORRUPT_DATA, txt, sizeof(txt));
+		return RTR_ERROR;
+	}
```

Comment:
this will never get executed

Reply:
turns out they did think about other bits being set in the flags bit field.. using a mask like 0x1 isn't necessary since the else-block is reached if `flags` is neither zero nor one meaning they're simply throwing an error if bits other than the 0x1 bit are set. In turn, to me that seems like a violation of this line in the draft

> Reserved fields (marked "zero" in PDU diagrams) MUST be zero on transmission and MUST be ignored on receipt.


The code's currently expecting the other bits to be zero rather than ignoring them.

Reply:
therefore: ignore them and use the mask